Fails to build with Clang 19.1.x

Trying to build vigra with clang 19.1.x fails because `multi_iterator_coupled.hxx` tries to modify the `this` pointer.

```
cd /home/bero/abf/vigra/BUILD/vigra-1.12.1-build/vigra-Version-1-12-1/build/src/impex && /usr/bin/c++ -DHasEXR -DHasHDF5 -DHasJPEG -DHasPNG -DHasTIFF -DHasZLIB -Dvigraimpex_EXPORTS -I/home/bero/abf/vigra/BUILD/vigra-1.12.1-build/vigra-Version-1-12-1/include -isystem /usr/include/OpenEXR -isystem /usr/include/Imath -pthread -ftemplate-depth-1024 -ftemplate-depth-1024 -std=gnu++11 -fPIC    -DH5_BUILT_AS_DYNAMIC_LIB -MD -MT src/impex/CMakeFiles/vigraimpex.dir/imageinfo.cxx.o -MF CMakeFiles/vigraimpex.dir/imageinfo.cxx.o.d -o CMakeFiles/vigraimpex.dir/imageinfo.cxx.o -c /home/bero/abf/vigra/BUILD/vigra-1.12.1-build/vigra-Version-1-12-1/src/impex/imageinfo.cxx
In file included from /home/bero/abf/vigra/BUILD/vigra-1.12.1-build/vigra-Version-1-12-1/src/impex/imageinfo.cxx:62:
In file included from /home/bero/abf/vigra/BUILD/vigra-1.12.1-build/vigra-Version-1-12-1/include/vigra/imageinfo.hxx:58:
In file included from /home/bero/abf/vigra/BUILD/vigra-1.12.1-build/vigra-Version-1-12-1/include/vigra/multi_iterator.hxx:43:
/home/bero/abf/vigra/BUILD/vigra-1.12.1-build/vigra-Version-1-12-1/include/vigra/multi_iterator_coupled.hxx:493:9: error: expression is not assignable
  493 |         --this;
      |         ^ ~~~~
1 error generated.
make[2]: *** [src/impex/CMakeFiles/vigraimpex.dir/build.make:219: src/impex/CMakeFiles/vigraimpex.dir/imageinfo.cxx.o] Error 1
```

--- a/include/vigra/multi_iterator_coupled.hxx
+++ b/include/vigra/multi_iterator_coupled.hxx
@@ -490,7 +490,7 @@ class CoupledScanOrderIterator<N, HANDLES, 0>
     CoupledScanOrderIterator operator--(int)
     {
         CoupledScanOrderIterator res(*this);
-        std::advance(this, -1);
+        std::advance(*this, -1);
         return res;
     }
 

--- a/include/vigra/multi_iterator_coupled.hxx
+++ b/include/vigra/multi_iterator_coupled.hxx
@@ -490,7 +490,7 @@ class CoupledScanOrderIterator<N, HANDLES, 0>
     CoupledScanOrderIterator operator--(int)
     {
         CoupledScanOrderIterator res(*this);
-        --this;
+        std::advance(this, -1);
         return res;
     }
 

